### PR TITLE
Update the authors data under `site/authors`

### LINKED ISSUE #108: Update the authors data under `site/authors`
### What feature do you want to see added?

Import the author's data from https://github.com/jenkins-infra/jenkins.io/tree/master/content/_data/authors and update here under `site/authors` in the repository.

The format is different, so if we can automate this with scripting will be a huge help.

For example https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/authors/KunalSin9h.adoc and https://github.com/jenkins-infra/docs.jenkins.io/blob/main/site/authors/KunalSin9h.adoc has different formats.

### PR DESCRIPTION
**Pull request by Nwaamaka-Iduwe**
Fixes #108

Automated Author Data Conversion for Gatsby Site

This pull request includes a script that automates the migration of author data from the `jenkins.io` repository format to the required `docs.jenkins.io` Gatsby format. Key details like name, GitHub, Twitter, LinkedIn, and biographies are extracted and reformatted accordingly. The script also ensures each author’s profile image file path is correctly updated to use the `.png` extension.

This automation helps streamline data migration, reduces manual entry errors, and supports easier future updates as new authors are added.

Please review and let me know if there are any adjustments needed!

### CODE REVIEW

**Review by krisstern — CHANGES_REQUESTED** [review]
Thanks @Nwaamaka-Iduwe for the PR! However, if you take a look at the https://github.com/jenkins-infra/docs.jenkins.io/tree/main/site/static/images/images/avatars, not all the images provided are in the `PNG` format, so some of the changes proposed in this PR will not work as a result. Could you please modify this script so that we would not change the image file name if it is given in the `JPEG` format with either the `.jpeg` or `.jpg` extensions? Please note some authors may not have provided us with an image at all. So we will need to consider those cases too.

**Comment by Nwaamaka-Iduwe** [comment]
Hi @krisstern okay, I'll work on this shortly. Apologies for the late response.

**Comment by krisstern** [comment]
No worries @Nwaamaka-Iduwe!

**Comment by Nwaamaka-Iduwe** [comment]
Hi @krisstern i made changes as requested. Not sure if you've had time to go over them.

**Review by krisstern — CHANGES_REQUESTED** [review]
Hi @Nwaamaka-Iduwe thanks for the updates! 

Basically we do not want any changes to existing profiles unless there is a very pressing reason unlike new updates to the existing profiles via PR's submitted by the user. 

Also, I am not sure why we are adding new lines like the below in the metadata section: for "site/authors/HrushikeshRao.adoc"

```
name: "Hrushikesh Rao"
github: "hrushi20"
twitter: "Hrushi2020"
linkedin: "hrushikesh-rao-7741311b0"
```

Could you plaase revert the unwanted changes?

**Comment by Nwaamaka-Iduwe** [comment]
Hi @krisstern thank you for your feedback. I've now updated the script so it doesn't include extra lines in the metadata. Could you please review when you can? Thank you!

**Review by krisstern — CHANGES_REQUESTED** [review]
Hi @Nwaamaka-Iduwe,

Thanks for the follow-up! 

However, I do not see any action regarding to my previous comment:
> Basically we do not want any changes to existing profiles unless there is a very pressing reason like new updates to the existing profiles via PR's submitted by the user.

So basically we do not need any change to the existing profiles. And for the fields where there is no data we do not want the value `N/A` either. You should just not have these fields instead.

File: .gitignore [hunk 1]
```diff
@@ -6,4 +6,4 @@ blog/public/
 blog/package-lock.json
 ui/build/ui-bundle.zip
 site/public/
-.DS_Store
\ No newline at end of file
+.DS_Storejenkins.io/
```
Comment (krisstern) [inline .gitignore:9]:
```suggestion
.DS_Store

```

Comment (krisstern) [reply .gitignore:9]: I don't think we are supposed to ignore the jenkins.io/ folder.

**Review by krisstern — CHANGES_REQUESTED** [review]

File: README.adoc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-= Jenkins Documentation
+git clone https://github.com/jenkins-infra/jenkins.io.git = Jenkins Documentation
```
Comment (krisstern) [inline README.adoc:1]:
```suggestion
= Jenkins Documentation
```

Comment (krisstern) [reply README.adoc:1]: This should not be changed

**Review by krisstern — CHANGES_REQUESTED** [review]
Hi @Nwaamaka-Iduwe, please revert all of the changes to existing `.adoc` files in the `site/authors` folder. We are happy with them and we do not want these to be modified in ways we had not originally planned.

**Comment by Nwaamaka-Iduwe** [comment]
Hi @krisstern I've made the changes as requested. Please let me know what else I can change and if the script itself solves the problem.

File: site/authors/KunalSin9h.adoc [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
-:page-author_name: Kunal Singh
-:page-github: KunalSin9h
-:page-twitter: KunalSin9h
-:page-linkedin: kunalsin9h
-:page-authoravatar: ../../images/images/avatars/KunalSin9h.jpg
+---
+---
 
-My name is Kunal and I'm currently pursuing a degree in Computer Science and Engineering. I do Full Stack Web Development, using technologies like React, NextJS, TailwindCSS,TypeScript and Go. I'm also passionate about Cloud computing, Backend and Databases.
+---
```
Comment (krisstern) [inline site/authors/KunalSin9h.adoc:4]: Pleaes undo the changes you have introduced to this and other **existing** files

File: site/authors/HrushikeshRao.adoc [hunk 4]
```diff
@@ -0,0 +1,12 @@
+---
+---
+
+---
+name: "Hrushikesh Rao"
+github: "hrushi20"
+twitter: "Hrushi2020"
+linkedin: "hrushikesh-rao-7741311b0"
+---
```
[the remark this answers is not in the document]
Comment (krisstern) [reply site/authors/HrushikeshRao.adoc:9]: Please follow the existing format and do not invent a new format from nowhere.

**Review by krisstern — CHANGES_REQUESTED** [review]
Hi @Nwaamaka-Iduwe I may need to close this PR as you have been having trouble following some very simple instructions on what kind of changes are needed. This PR as it stands is unacceptable and you have not reverted the changes I already told you a few times previously not to make.

**Comment by krisstern** [comment]
Hi @Nwaamaka-Iduwe I have already taken the steps to clarify by making the suggested changes for you in the case of a new `.adoc` file. But for the existing files please **revert** all your changes. The way the `.adoc` files work is that they have been configured to look for a pattern here, so you will need something like the following:

```AsciiDoc
= About the Author
:page-layout: author
:page-author_name: Mark Waite
:page-twitter: MarkEWaite
:page-github: markewaite
:page-irc: markewaite
:page-linkedin: markwaite
:page-authoravatar: ../../images/images/avatars/markewaite.jpg

Mark is a member of the link:/project/board/[Jenkins governing board], a long-time Jenkins user and contributor, a core maintainer, and maintainer of the plugin:git[git plugin], the plugin:git-client[git client plugin], the plugin:platformlabeler[platform labeler plugin], the plugin:embeddable-build-status[embeddable build status plugin], and several others. He is one of the authors of the link:/doc/developer/tutorial-improve/["Improve a plugin"] tutorial.
// He is active in link:/sigs/[Jenkins special interest groups] including the link:/sigs/docs/[Docs SIG], link:/sigs/platform[Platform SIG], link:/sigs/ux[User Experience SIG], and link:/sigs/advocacy-and-outreach[Advocacy SIG].
// He has mentored Google Summer of Code projects including link:/projects/gsoc/2022/projects/automatic-git-cache-maintenance/[automatic git cache maintenance (2022)], link:/projects/gsoc/2021/projects/git-credentials-binding/[git credentials binding (2021)], and link:/projects/gsoc/2020/projects/git-plugin-performance/[git plugin performance improvements (2020)].
```

The header part is supposed to follow a strict template without any double quotes and contains the meta-data:
```AsciiDoc
= About the Author
:page-layout: author
:page-author_name: Firstname Lastname
:page-twitter: twitter-handle
:page-github: github-username
:page-irc: gitter-handle
:page-linkedin: linkedin-handle
:page-authoravatar: ../../images/images/avatars/image-filename.jpg
```

The line starting with `=` is for the title. 

If you are having problems inferring instructions from existing data, then maybe it is better for me to close this PR and let someone else finish it. It is supposed to be a very easy "good first issue" to work on. Also, you should be asking me clarifying questions since I don't know what and why you don't understand about the issue.

**Comment by Nwaamaka-Iduwe** [comment]
Hi @krisstern 

I’ve now made the necessary changes to address your feedback:
	1.	The existing files in the repository remain unaltered, as requested.
	2.	The script correctly converts the source author data to the required format, ensuring headings like :page-author_name, :page-twitter, and others are applied accurately.
	3.	Missing fields are handled cleanly without placeholders like “N/A,” keeping the output neat and consistent.
	4.	The converted files have been added only for new authors, aligning with the specified requirements.

I’d also like to share that some of the feedback in your earlier message were dismissive, and I think open source works best when we foster a learning environment. I’m not a developer or software engineer by training, so clearer guidance earlier on would have helped me understand your expectations more quickly and move forward faster.

I value collaboration and learning, so I hope we can maintain a constructive and supportive tone as we continue working together. Please let me know if there’s anything else you’d like me to improve.

**Comment by krisstern** [comment]
Hi @Nwaamaka-Iduwe,
Thanks for your updates! I will review your PR shortly. 

Regarding your comments
> I’d also like to share that some of the feedback in your earlier message were dismissive, and I think open source works best when we foster a learning environment. I’m not a developer or software engineer by training, so clearer guidance earlier on would have helped me understand your expectations more quickly and move forward faster.

I think my tone was justified as you did seem to have some issues following some very simple/basic instructions. As I have repeated myself **at least three times on the same request** which seem to have been ignored, maybe you should also consider your competency and make improvements on your own. As a repo maintainer it is really not my responsibility to make sure you learn or grow. In open-source we do strive to foster a collaborative environment, but that does not include free mentorship necessarily. I do hope you can understand that.

File: authors_conversion.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    # Fetch and convert the file content
+    source_content = fetch_remote_file(source_url)
+    if source_content:
+        converted_content = convert_author_data(source_content)
+
+        # Write the new file to the target directory
+        with open(target_file, "w", encoding="utf-8") as f:
+            f.write(converted_content)
+
+print("All new author files have been added without altering existing files.")
```
Comment (krisstern) [inline authors_conversion.py:101]:
```suggestion
print("All new author files have been added without altering existing files.")

```

Comment (krisstern) [reply authors_conversion.py:101]: You do need an extra blank line for the EOF here

**Review by krisstern — CHANGES_REQUESTED** [review]
All the new `site/authors` `.adoc` files are missing the `:page-authoravatar:` meta-data values. This should not be the case, as I know for a fact some profiles do have images included. Please be aware this part may be more involved as the extension of each avatar file is not standardized.

**Comment by Nwaamaka-Iduwe** [comment]
Hi @krisstern thanks for the suggestion. Please let me know if the new commit meets the requirements using glob.

File: authors_conversion.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+            else:
+                print(f"Failed to fetch content for {author_file}: HTTP {response.status_code}")
+        except Exception as e:
+            print(f"Error processing {author_file}: {e}")
+
+    print("Processing complete.")
+
+
+if __name__ == "__main__":
+    main()
```
Comment (krisstern) [inline authors_conversion.py:159]:
```suggestion
    main()

```

Comment (krisstern) [reply authors_conversion.py:159]: For most non-config files, we usually have a blank line in the EOF.

**Review by krisstern — CHANGES_REQUESTED** [review]
Hi @Nwaamaka-Iduwe, this PR is looking really nice now! I appreciate you being perceptive to my suggestions. Except for one minor change I think we can merge this soon.

**Review by krisstern — APPROVED** [review]
Appreciate the PR @Nwaamaka-Iduwe!

**Comment by Nwaamaka-Iduwe** [comment]
Thanks @krisstern 
Looking forward to contributing more after the holidays :)

**Comment by krisstern** [comment]
You are welcome @Nwaamaka-Iduwe! I am impressed by how fast you have improved.

**Comment by krisstern** [comment]
@Nwaamaka-Iduwe Please do not change the existing filenames next time.

**Pull request merged**